statsmodels/stats/contingency_tables.py

```diff
@@ -0,0 +1,1312 @@
+        The estimated row and column marginal distributions.
+    independence_probabilities : array-like
+        Estimated cell probabilities under independence.
+    fittedvalues : array-like
+        Fitted values under independence.
+    pearson_resids : array-like
+        The Pearson residuals.
+    standardized_resids : array-like
+        Residuals with approximately unit variance.
+    chi2_contribs : array-like
+        The contribution of each cell to the chi^2 statistic.
+    local_logodds_ratios : array-like
+        The local log odds ratios are calculated for each 2x2 subtable
+        formed from adjacent rows and columns.
+    local_oddsratios : array-like
+        The local odds ratios are calculated from each 2x2 subtable
+        formed from adjacent rows and columns.
+    cumulative_log_oddsratios : array-like
+        The cumulative log odds ratio at a given pair of thresholds is
+        calculated by reducing the table to a 2x2 table based on
+        dichotomizing the rows and columns at the given thresholds.
+        The table of cumulative log odds ratios presents all possible
+        cumulative log odds ratios that can be formed from a given
+        table.
+    cumulative_oddsratios : array-like
+        The cumulative odds ratios are calculated by reducing the
+        table to a 2x2 table based on cutting the rows and columns at
+        a given point.  The table of cumulative odds ratios presents
+        all possible cumulative odds ratios that can be formed from a
+        given table.
+
+    See also
+    --------
+    statsmodels.graphics.mosaicplot.mosaic
+    scipy.stats.chi2_contingency
+
+    Notes
+    -----
+    The inference procedures used here are all based on a sampling
+    model in which the units are independent and identically
+    distributed, with each unit being classified with respect to two
+    categorical variables.
+    """
+
+    def __init__(self, table, shift_zeros=True):
+
+        self.table_orig = table
+        self.table = np.asarray(table, dtype=np.float64)
+
+        if shift_zeros and (self.table.min() == 0):
+            self.table = self.table + 0.5
+
+
+    @classmethod
+    def from_data(cls, data, shift_zeros=True):
+        """
+        Construct a Table object from data.
+
+        Parameters
+        ----------
+        data : array-like
+            The raw data, from which a contingency table is constructed
+            using the first two columns.
+        shift_zeros : boolean
+            If True and any cell count is zero, add 0.5 to all values
+            in the table.
+
+        Returns
+        -------
+        A Table instance.
+        """
+
+        if isinstance(data, pd.DataFrame):
+            table = pd.crosstab(data.iloc[:, 0], data.iloc[:, 1])
+        else:
+            table = pd.crosstab(data[:, 0], data[:, 1])
+
+        return cls(table, shift_zeros)
+
+
+    @cache_readonly
+    def nominal_association(self):
```

Comment:
two possible problems

- nominal_association might mean a measure for how large the association is (eg. and agreement measure or effectsize), maybe prefix `test_` ?
- making this a cached attribute is not future proof, if for example we want to use likelihood ratio (Q) test or general powerdivergence (Cressie-Reed), or exact tests. 

(I don't remember where I mentioned this before)